File: Python/Greeting/GreetingServer/Services/GreetingService.py
import greeting_pb2
import greeting_pb2_grpc

class GreetingService(greeting_pb2_grpc.GreetingServiceServicer):

    def Greet(self, request, context):
        return greeting_pb2.GreetingResponse(message = f'Hello, {request.sender_name}.')

File: .NET Core 3.0/Greeting/GreetingServer/Services/GreetingService.cs
using Generated;
using GreetingServer.Utils.Messages;
using Grpc.Core;
using NLog;
using System.Threading.Tasks;

namespace GreetingServer.Services
{
    internal sealed class GreetingService : Generated.GreetingService.GreetingServiceBase
    {
        #region Logger
        private static Logger Logger { get; } = LogManager.GetCurrentClassLogger();
        #endregion

        public override Task<GreetingResponse> Greet(GreetingRequest request, ServerCallContext context)
        {
            //Logger.Info(LoggerMessages.ServiceRequestMessage(service: nameof(Greet), host: context.Host, peer: context.Peer));
            return Task.FromResult(new GreetingResponse() { Message = $"Hello, {request.SenderName}." });
        }
    }
}
